### DISCUSSION TITLE
Dirichlet condition which is a function

### DISCUSSION THREAD

**Discussion by Toomyalla** [firedrakeproject/firedrake#2565] [Firedrake support] [answered]
I am trying to  solve Poisson's equationwith  non trivial boun dary conditions  but it does not work .  I need help how to code the conditions and this is my attempt. 

```python
from firedrake import *

mesh = UnitSquareMesh(10, 10)

V = FunctionSpace(mesh, "CG", 1)

u = TrialFunction(V)
v = TestFunction(V)

a = (inner(grad(u), grad(v)) + inner(u, v)) * dx
L = Constant(0)*v*dx

x =SpatialCoordinate(mesh)

g = Function(V)
g.interpolate(min(4*x[0],4-4*x[0]))
bc1=DirichletBC(V, g ,1)

boundary_ids={2, 3, 4}
bc2=DirichletBC(V,0, boundary_ids)

u = Function(V)

solve(a == L, u,bcs=[bc1,bc2], solver_parameters={'ksp_type': 'cg', 'pc_type': 'none'})
```

Thank you in advance,

**Comment by dham** [comment]
Thanks for the question. I notice a couple of things immediately:
* The equation being solved is Helmholtz, not Poisson. Is this intentional?
* For a small problem like this, changing the default solver parameters probably isn't desirable.

You haven't specified what you mean by "it does not work". Could you please detail what the observed output is and what you expect.

**Comment by Toomyalla** [comment]
Sorry, I mean a to be
a = dot(grad(u), grad(v))* dx

I get the error 
UFL:ERROR UFL conditions cannot be evaluated as bool in a Python context.
Traceback (most recent call last):
  File "/home/firedrake/mycode/poisson.py", line 26, in <module>
    g.interpolate((min(4*y,4-4*y)))
  File "/home/firedrake/firedrake/src/ufl/ufl/conditional.py", line 34, in __bool__
    error("UFL conditions cannot be evaluated as bool in a Python context.")
  File "/home/firedrake/firedrake/src/ufl/ufl/log.py", line 158, in error
    raise self._exception_type(self._format_raw(*message))
ufl.log.UFLException: UFL conditions cannot be evaluated as bool in a Python context.

I expect an answer of approximately 0.325 when I ask for the solution at the center.
print(u.at([0.5,0.5]))

**Comment by ksagiyam** [comment, marked as answer]
I think you want `g.interpolate(conditional(4*x[0] > 4-4*x[0], 4-4*x[0], 4*x[0]))` instead of `g.interpolate(min(4*x[0],4-4*x[0]))`.

**Comment by Toomyalla** [comment]
Thank you. It is solved but without the use of the index 0. 

g.interpolate(conditional(4*y > 4-4*y, 4-4*y, 4*y))
